src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,188 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        this.head = null;
+        this.tail = null;
+        this.size = 0;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value, null, tail);
+        if (tail == null) {
+            head = newNode;
+            tail = newNode;
+        } else {
+            tail.setNext(newNode);
+            newNode.setPrev(tail);
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index out of bounds");
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> newNode;
+        if (index == 0) {
+            newNode = new Node<>(value, head, null);
+            head.setPrev(newNode);
+            head = newNode;
+        } else {
+            Node<T> current = getNode(index);
+            newNode = new Node<>(value, current, current.getPrev());
+            current.getPrev().setNext(newNode);
+            current.setPrev(newNode);
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
```

Comment:
Consider adding a null check for the `list` parameter to prevent a `NullPointerException`. This can be done by adding a condition like `if (list == null) throw new NullPointerException("List cannot be null");`.